src/config.rs

```diff
@@ -112,3 +124,48 @@ impl AsRef<Client> for Config {
         &self.client
     }
 }
+
+#[non_exhaustive]
+#[derive(thiserror::Error, Debug, Clone, Copy)]
+pub enum SessionSecretsDecodeError {
+    #[error("session secret must be at least 32 bytes after base64 decode")]
+    TooShort,
+    #[error("session secret must be url-safe no-pad base64")]
```

Comment:
nit: standard nomenclature
```suggestion
    #[error("session secret must be base64url without padding")]
```